File: restaurant_backend/src/main/java/model/Dish.java
package model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Dish {

    @Id
    private String id;
    private String name;
    private String type;
    private String category;
    private String[] ingredients;
    private Integer maxQuantity;
    private Double price;
    private String imageUrl;
}

File: restaurant_frontend/src/app/model/dish.ts
export class Dish{
    id: string;
    name: string;
    type: string;
    category: string;
    ingredients: Array<string>;
    maxQuantity: number;
    price: number;
    imageUrl: string;
    selectedQuantity: number;

    constructor(id: string, name: string, type: string, category: string, ingredients: Array<string>, maxQuantity: number, price: number, imageUrl: string) {
        this.id = id;
        this.name = name;
        this.type = type;
        this.category = category;
        this.ingredients = ingredients;
        this.maxQuantity = maxQuantity;
        this.price = price;
        this.imageUrl = imageUrl;
        this.selectedQuantity = 0;
    }
}